### PR TITLE
Fix find command

### LINKED ISSUE #124: [PE-D][Tester B] App silently errors out when using find function with a name containing multiple words
How to reproduce:
1. Open the app
2. Create user with name "john doe", among other users.
2. Enter command: 

> find n/john doe

Expected outcome:
List of people with name "john doe"

Actual outcome:
The command result does not show anything, people john doe is not found, and there is an error message in the terminal.

Error:

Before command:

After command:

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: ReflectiveObsidian/ped#2

### LINKED ISSUE #126: [PE-D][Tester E] Find description does not work properly
If I have contacts with `Suffers from xxxxx`, I have inputted `find d/Suffers from `. This tells me that 4 contacts are found but with no contacts shown at all. 

-------------
Labels: `type.FunctionalityBug` `severity.High`
original: starsia/ped#5

### LINKED ISSUE #127: [PE-D][Tester E] Find description command has no error message
There is no message that tells me that this description is not found. 

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: starsia/ped#6

### LINKED ISSUE #129: [PE-D][Tester E] Find description command does not parse my input correctly
If there are spaces in the find command, despite it having keywords in multiple contacts, `find d/suffers    from`, no error message is thrown and nothing happens. 

-------------
Labels: `type.FunctionalityBug` `severity.High`
original: starsia/ped#7

### LINKED ISSUE #130: [PE-D][Tester E] Find description command crashes despite complete match
I have ran the command `find d/suffers from blood disorder`. However, there is no filtering done, and the contact list also disappears entirely. 

original contact list:

bug screenshot:

-------------
Labels: `type.FunctionalityBug` `severity.High`
original: starsia/ped#8

### LINKED ISSUE #131: [PE-D][Tester C] find command allows user to type a random premable which results in no results being displayed
When using the `find` command, it is possible for the user to type in a string of at least 3 characters after a space and the command successfully parses, albeit returning no person.

e.g. `find mari` results in this response

You might want to give a error message regarding these types of input

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: RingoftheKing/ped#3

### LINKED ISSUE #132: [PE-D][Tester E] General find command works but gives 0 output without the use of flags
If I type `find alex`, `find 1234` or `find abc123`, I get the error message `0 persons listed`. 

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: starsia/ped#9

### LINKED ISSUE #143: [PE-D][Tester C] find command fails when using the prefix d/ and k/
When using the `find` command, attempting to use the `d/` or `k/` prefix will not work nor give any response to the user. 

e.g. entering `find d/blood` doesn't give any response.

 

e.g. entering `find k/Choon Yu` doesn't give any response

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: RingoftheKing/ped#4

### LINKED ISSUE #144: [PE-D][Tester C] find command allows user to type non existent tags
The user may type an incorrect prefix such as `x/`  or `y/` and the find command parses successfully albeit returning no persons.

e.g. user types `find x/ y/ z/`

user might expect some form of error message but ConnectCare instead shows an empty list

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: RingoftheKing/ped#5

### LINKED ISSUE #147: [PE-D][Tester D] Find feature allowing for wrong prefixes or no prefixes at all
Find command allows for wrong prefixes. Special characters used does not show a bug but instead leads to no result.
 e.g "find * a", "find / a", "find . a", "find /n doe" are allowed.

-------------
Labels: `type.FeatureFlaw` `severity.High`
original: Xilef121/ped#3

### LINKED ISSUE #160: [PE-D][Tester E] Crashes upon `find n/Alex Yeoh` command
A complete match of an existing contact using find command `find n/alex yeoh` or `find n/Alex Yeoh` crashes the contact base temporarily and yields no finds. Upon recovery using `list` command, `find n/alex yeoh` is unable to be found in past commands using the up arrow key. 

-------------
Labels: `severity.High` `type.FeatureFlaw`
original: starsia/ped#18

### PR DESCRIPTION
**Pull request by eliotlee68**
Closes #160 , closes #147 , closes #144 , closes #143, closes #132, closes #131,  closes #130 , closes #129 , closes #127 , closes #126, closes #124 .

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 1]
```diff
@@ -58,4 +63,14 @@ public FindCommand parse(String args) throws ParseException {
                 new DescriptionContainsKeywordsPredicate(descriptionKeywords));
     }
 
+    private void checkForNulls(List<String>... lists) throws ParseException {
+        for (List<String> list : lists) {
+            for (String str: list) {
+                if (str.trim().isEmpty()) {
+                    throw new ParseException(
+                            String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+                }
```
Comment (pjanthony2001) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java:72]: code quality use streams instead, map, filter and then anyMatch. Then an if statemen to throw an errror

File: src/main/java/seedu/address/commons/util/StringUtil.java [hunk 2]
```diff
@@ -30,17 +29,23 @@ public static boolean containsWordIgnoreCase(String sentence, String word) {
...
+                    wordFound = true;
+                    break;
+                }
+            }
+            if (!wordFound) {
+                return false;
             }
         }
-        return false;
+        return true;
```
Comment (pjanthony2001) [inline src/main/java/seedu/address/commons/util/StringUtil.java:48]: Use streams, too much nesting is an issue in code quality.

**Review by pjanthony2001 — COMMENTED** [review]
Heyo mate, remember to add the issues you fixed in your description.

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 3]
```diff
@@ -31,6 +31,8 @@ public class FindCommandParser implements Parser<FindCommand> {
      * and returns a FindCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
      */
+
+    @SuppressWarnings("unchecked")
```
Comment (pjanthony2001) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java:35]: is there a reason for unchecked cast?

Comment (eliotlee68) [reply src/main/java/seedu/address/logic/parser/FindCommandParser.java]: leftover from previous code, deleted

**Comment by pjanthony2001** [comment]
LGTM

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 4]
```diff
@@ -82,4 +78,13 @@ private void checkForNulls(List<String>... lists) throws ParseException {
                 String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
     }
+
+    private void checkAllEmpty(List<String>... lists) throws ParseException {
+        boolean allEmpty = Arrays.stream(lists)
```
Comment (Jolonauh) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java:83]: boolean should be something like isAllEmpty

**Review by Jolonauh — COMMENTED** [review]
Minor issue

**Comment by pjanthony2001** [comment]
LGTM

**Pull request merged**